numpy_financial/_financial.py

```diff
@@ -124,11 +124,28 @@ def fv(rate, nper, pmt, pv, when='end'):
 
     """
     when = _convert_when(when)
-    (rate, nper, pmt, pv, when) = map(np.asarray, [rate, nper, pmt, pv, when])
-    temp = (1+rate)**nper
-    fact = np.where(rate == 0, nper,
-                    (1 + rate*when)*(temp - 1)/rate)
-    return -(pv*temp + pmt*fact)
+    rate, nper, pmt, pv, when = np.broadcast_arrays(rate, nper, pmt, pv, when)
+
+    fv_array = np.empty_like(rate)
+    zero = rate == 0
+    nonzero = ~zero
+
+    fv_array[zero] = -(pv[zero] + pmt[zero] * nper[zero])
+
+    rate_nonzero = rate[nonzero]
+    nper_nonzero = nper[nonzero]
```

Comment:
I can't see ``nper_nonzero`` being used anywhere else, have I missed it being used anywhere?